### ISSUE TITLE
[AXE-CORE]: VAMC - Links must have discernible text

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#21791]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix issue.
1. At message the accessibility specialist when issue is resolved.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement
Accessibility specialists discovered Nightwatch axe checks failing silently during a recent staging review. Multiple axe violations were discovered where empty link tags were on the pages. These are problematic because screen readers will announce "Link" but not what the link is. 

## Tasks
- [ ] If there is no value or HREF, the link `<a></a>` should be removed. 

## Environments

<details>
<summary> The following received errors due to empty mental health phone numbers </summary>

![Screen Shot 2021-03-19 at 2 42 51 PM](https://user-images.githubusercontent.com/934879/111834431-98e2d400-88c1-11eb-9411-89ea9cb32d8c.png)

- [x] https://www.va.gov/durham-health-care/locations/clayton-east-raleigh-va-clinic/
- [x] https://www.va.gov/altoona-health-care/locations/huntingdon-county-va-clinic/
- [ ] https://www.va.gov/butler-health-care/locations/butler-va-medical-center/
- [ ] https://www.va.gov/butler-health-care/locations/clarion-county-va-clinic/
- [ ] https://www.va.gov/butler-health-care/locations/cranberry-township-va-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/cheyenne-va-mobile-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/laramie-va-mobile-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/rawlins-va-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/sidney-va-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/sterling-va-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/torrington-va-mobile-clinic/
- [ ] https://www.va.gov/cheyenne-health-care/locations/wheatland-va-mobile-clinic/
- [ ] https://www.va.gov/durham-health-care/locations/brier-creek-va-clinic/
- [ ] https://www.va.gov/durham-health-care/locations/durham-county-va-clinic/
- [ ] https://www.va.gov/durham-health-care/locations/hillandale-road-va-clinic/
- [ ] https://www.va.gov/durham-health-care/locations/raleigh-iii-va-clinic/
- [ ] https://www.va.gov/durham-health-care/locations/wake-county-va-clinic/
- [ ] https://www.va.gov/eastern-oklahoma-health-care/locations/ernest-childers-department-of-veterans-affairs-outpatient/
- [ ] https://www.va.gov/eastern-oklahoma-health-care/locations/mccurtain-county-va-clinic/
- [ ] https://www.va.gov/eastern-oklahoma-health-care/locations/muskogee-east-va-clinic/
- [ ] https://www.va.gov/eastern-oklahoma-health-care/locations/yale-avenue-va-clinic/
- [ ] https://www.va.gov/erie-health-care/locations/mckean-county-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/anaconda-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/browning-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/cut-bank-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/dr-joseph-medicine-crow-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/glasgow-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/hamilton-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/helena-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/lewistown-va-clinic/
- [ ] https://www.va.gov/montana-health-care/locations/miles-city-va-community-living-center/
- [ ] https://www.va.gov/montana-health-care/locations/plentywood-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/altus-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/blackwell-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/clinton-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/enid-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/fourteenth-street-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/lawton-north-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/norman-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/north-may-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/tinker-va-clinic/
- [ ] https://www.va.gov/oklahoma-city-health-care/locations/yukon-va-clinic/
- [ ] https://www.va.gov/philadelphia-health-care/locations/camden-va-clinic/
- [ ] https://www.va.gov/philadelphia-health-care/locations/chestnut-street-va-clinic/
- [ ] https://www.va.gov/philadelphia-health-care/locations/fourth-street-va-clinic/
- [ ] https://www.va.gov/philadelphia-health-care/locations/philadelphia-va-domiciliary/
- [ ] https://www.va.gov/philadelphia-health-care/locations/west-philadelphia-va-clinic/
- [ ] https://www.va.gov/salt-lake-city-health-care/locations/elko-va-clinic/
- [ ] https://www.va.gov/salt-lake-city-health-care/locations/idaho-falls-va-clinic/
- [ ] https://www.va.gov/salt-lake-city-health-care/locations/price-va-clinic/
- [ ] https://www.va.gov/salt-lake-city-health-care/locations/roosevelt-va-clinic/
- [ ] https://www.va.gov/sheridan-health-care/locations/afton-va-clinic/
- [ ] https://www.va.gov/sheridan-health-care/locations/evanston-va-clinic/
- [ ] https://www.va.gov/sheridan-health-care/locations/worland-va-clinic/
- [ ] https://www.va.gov/western-colorado-health-care/locations/glenwood-springs-va-clinic/
- [ ] https://www.va.gov/western-colorado-health-care/locations/grand-junction-va-mobile-clinic/
- [ ] https://www.va.gov/western-colorado-health-care/locations/major-william-edward-adams-department-of-veterans-affairs/
- [ ] https://www.va.gov/western-colorado-health-care/locations/moab-va-clinic/
- [ ] https://www.va.gov/western-colorado-health-care/locations/montrose-va-clinic/
- [ ] https://www.va.gov/wilkes-barre-health-care/locations/northampton-county-va-clinic/
- [ ] https://www.va.gov/wilkes-barre-health-care/locations/tobyhanna-va-clinic/
- [ ] https://www.va.gov/wilkes-barre-health-care/locations/wayne-county-va-clinic/

</details>

---

<details>
<summary> The following received errors due to empty main health phone numbers </summary>

- [ ] https://www.va.gov/salt-lake-city-health-care/locations/

</details>

---

<details>
<summary> The following received errors due to empty main and mental health phone numbers </summary>

- [ ] https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/

</details>

---
The issue observed on https://www.va.gov/eastern-oklahoma-health-care/news-releases/eastern-oklahoma-va-begins-covid-19-vaccine-distribution/ has been moved to #22139

---

The issue observed on https://www.va.gov/outreach-and-events/events/accessing-covid-19-vaccinations-at-the-va-what-you-need-to-know/ has been moved to #22138 

AC
Phone number labels should only be present with a phone number on location list and on the facility pages.

**Comment by mmiddaugh** [comment]
This one should be resolved by https://github.com/department-of-veterans-affairs/va.gov-team/issues/21073

**Comment by mmiddaugh** [comment]
Per @1Copenut #21073 is specific to VAMCs' /location URL and template. 
This issue (#21791) is a collection of all the links missing discernible text that might span across one or more templates.

**Comment by mmiddaugh** [comment]
A fix (#21073) to resolve this issue has been merged and is available to be validated on Prod @1Copenut

**Comment by mmiddaugh** [comment]
Mental health phone numbers are resolved but Main phone numbers are still empty. Seen on https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/ 
![image](https://user-images.githubusercontent.com/55411834/112537246-1308d200-8d74-11eb-8ecb-0bc082a8a1d2.png)

**Comment by mmiddaugh** [comment]
If there is no phone number, do not show label.
Also need to communicate to Lighthouse that facilities are sent without main phone

**Comment by mmiddaugh** [comment]
The missing phone number is for a location which is not yet open and should not have been published. This will be addressed by unpublishing this location (VAMC upgrade team). This changes defect level but we will also ensure we fail more gracefully in the future.

**Comment by mmiddaugh** [comment]
This location has been archived by the VAMC team and the defect should be resolved.

**Comment by 1Copenut** [comment]
@mmiddaugh I went through the list of updates this morning. There are 4 pages with links still missing discernible text:
* https://www.va.gov/salt-lake-city-health-care/locations/
* https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/
* https://www.va.gov/eastern-oklahoma-health-care/news-releases/eastern-oklahoma-va-begins-covid-19-vaccine-distribution/
* https://www.va.gov/outreach-and-events/events/accessing-covid-19-vaccinations-at-the-va-what-you-need-to-know/

One of them looks like an Outreach and Events page, so outside VAMCs. The other three are for Salt Lake and Eastern Oklahoma. Let me know if we should re-open this ticket or create a new one to close those out. Thank you!

**Comment by davidconlon** [comment]
@1Copenut I think we might want to close this and have other issues(see below)
Fundamentally, How does VA.gov want to scale and be reporting 508 issues to VAMC and (soon to be) Tier 3 content producers who won't understand github, etc.
For the short term, we should assign the VAMC to @SgardnerPMP and VAMC board as I would assume the preponderance of issues will be creeping up there (once these are resolved) but we need something scalable by EOY because Stan's team will likely have moved on to other work.

https://www.va.gov/salt-lake-city-health-care/locations/
https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/
We tried to archive the locations but this failed and CMS has something in their backlog to fix that part
Being fixed as part of: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/21404

https://www.va.gov/eastern-oklahoma-health-care/news-releases/eastern-oklahoma-va-begins-covid-19-vaccine-distribution/
was a content issue from the field. I've gone in and fixed it and will publish shortly
How can we move forward in creating a 508 training for content producers to mitigate problems?
What's the right way to report/track/resolve errors produced by external content producers? (and how do we hound the people who are non-responsive, etc. big problem space)
I'm looking to Platform for some guidance on that because I think there have been conversations about this (or maybe I'm interpreting some stuff I've heard in meetings incorrectly)

Outreach and Events issue should be an issue for Public Websites.
